libbeat/tests/docker/docker.go

```diff
@@ -43,7 +43,7 @@ func NewClient() (Client, error) {
 }
 
 // ContainerStart pulls and starts the given container
-func (c Client) ContainerStart(image string, cmd []string, labels map[string]string) (string, error) {
+func (c Client) ContainerStart(image string, cmd []string, labels map[string]string, hostConfig *container.HostConfig) (string, error) {
```

Comment:
Nit. Consider defining new method to keep backwards compatibility. Though probably nothing is using this method outside of beats.
```suggestion
// ...
type StartOptions struct {
    HostConfig *container.HostConfig
}

// ...
func ContainerStart(image string, cmd []string, labels map[string]string) (string, error) {
    return ContainerStartWithOptions(image, cmd, labels, StartOptions{})
}

// ...
func (c Client) ContainerStartWithOptions(image string, cmd []string, labels map[string]string, options StartOptions) (string, error) {
...
```

Reply:
Thank you! This test is not correct nevertheless so I will revert this. Unfortunately we don't have proper tests at the moment that I could extend to cover this feature addition.